code cleanup

fuzzard opened the pull request:
Further cleanup of existing code. Removing unused variables/functions.
Stub OpenLiveStream and CloseLiveStream as they are not required for implementation as we now rely on GetChannelStreamProperties
Removed some globals that are not used throughout the codebase anymore.

Let me know if ive overstepped anything at this stage. I think this is the final pass for cleanups that i can see at this stage.

ksooo commented on src/HDHomeRunTuners.cpp:
```diff
@@ -395,15 +389,9 @@ std::string HDHomeRunTuners::_GetChannelStreamURL(int iUniqueId)
-      }
-    }
-  }        
+        return jsonChannel["URL"].asString();
```
wrong indentation?

ksooo commented on src/HDHomeRunTuners.h:
```diff
@@ -65,7 +65,7 @@ class HDHomeRunTuners
 
 public:
-  HDHomeRunTuners();
+  HDHomeRunTuners(){};
```
Looks ugly with a blank between `)` and `{`.

ksooo commented on src/Utils.h:
```diff
@@ -47,7 +47,6 @@ int DbgPrintf(const char* szFormat, ...);
   } while (0)
 
 #define PVR_STRCPY(dest, source) do { strncpy(dest, source, sizeof(dest)-1); dest[sizeof(dest)-1] = '\0'; } while(0)
```
Maybe you can also get rid of this ugly (and unneeded) macro?

ksooo commented on src/client.h:
```diff
@@ -27,6 +27,8 @@
 
+using namespace ADDON;
+
 typedef std::string String;
```
Another ugly piece of code...

fuzzard commented:
Removed the typedef and the macro as suggested.
The indentation is correct, i can put the {} back in if you think that makes it easier to read. The github diff makes it look a lot more complicated than the change is.

Happy to squash, unless you can do that on merge, just let me know

fuzzard commented:
rebased and added changelog/version bump.
Cheers

fuzzard commented:
squashed and rebased after jsoncpp merge.
Also did some changes to headers. Explicitly putting the header includes in files that use them, rather than having them inherit from other headers. Reordered to suit kodi guidelines as well.
also had another unused variable thats been removed.

fuzzard commented:
Couple more minors changed.
Removed addon namespace usage as it was only used for logging for the most part, so have just directly used the ADDON::XXXX types
Further header alterations. Make sure each file imports any and all headers required. In the past a lot of headers were just being pulled in due to the inclusion of the p8-platform headers. 
Shuffled some variable declarations to better suit when they are required.
Removed another typdef that was only used a single time. Clearer to just have the declaration as it is i think.

fuzzard commented:
@ksooo any chance of a final review. Ive repushed all of the changes and split it into individual commits to make it easier to review. The single commit was getting a bit out of hand to keep track of everything i think.

I also had a relook and think about the PVR_STRCPY define, and i didnt see it before, but it null terminates whatever is passed to it. In my initial pass of removing that, i wasnt dealing with null termination at all, which obviously is an issue. Should i stick with the define, or is the strcpy + null terminate at each use ok from a readability viewpoint?

fuzzard commented:
Another commit added. Currently a discover is never done past initial startup of addon. This means no additional tuners will ever be added unless the addon is restarted. The change will now add a tuner without a full discover if it is not found in the m_Tuners vector. To handle a tuner disappearing, it will do a clear on the vector if the size is larger than the discovered nTunerCount returned by the discovery request.
This doesnt really handle the situation of the tuner count staying the same, but with different tuners. This is such a corner case, that i couldnt work out a clean way to handle it without a lot of extra code for no real use. On the subsequent update call (60 minutes), this should all wash out as the m_tuners size will be larger than the next discovery, so it will reset the vector and recreate it all anyway.

Figured id throw this and 2 other commits into this PR, as its proven difficult to continue to get PR's merged (this is in no way aimed at you ksooo, because i know you have far more stuff to do than keep looking at one measly addon). 
If required i can pull out the feature change commits and put into another PR if desired.

ksooo commented:
> Another commit added. 

 Hold on! Give me some time to review the PR. I have the feeling if I find no time soon I will be confronted with a 50 commits PR containing all kind of different stuff. ;-)

ksooo commented on pvr.hdhomerun/addon.xml.in:
```diff
@@ -1,7 +1,7 @@
 <addon
   id="pvr.hdhomerun"
-  version="3.1.9"
+  version="3.1.10"
```
If you add new features, please increase minor version number (3.1.9 => 3.2.0 in this case).

ksooo commented on pvr.hdhomerun/changelog.txt:
```diff
@@ -1,3 +1,8 @@
+v3.1.10
+- code cleanup
+- guide filters updated
+- additional epg_tag fields added
```
If you add new features, please increase minor version number (3.1.9 => 3.2.0 in this case).

ksooo commented on src/client.cpp:
```diff
@@ -258,67 +253,59 @@ PVR_ERROR GetChannels(ADDON_HANDLE handle, bool bRadio)
-  
+  strncpy(signalStatus.strAdapterName, "PVR HDHomeRun Adapter 1",
+          sizeof(signalStatus.strAdapterName) - 1);
+  signalStatus.strAdapterName[sizeof(signalStatus.strAdapterName)-1] = '\0';
```
Could you please add blanks around the `-`?

ksooo commented on src/client.cpp:
```diff
@@ -258,67 +253,59 @@ PVR_ERROR GetChannels(ADDON_HANDLE handle, bool bRadio)
+  signalStatus.strAdapterName[sizeof(signalStatus.strAdapterName)-1] = '\0';
+  strncpy(signalStatus.strAdapterStatus, "OK",
+          sizeof(signalStatus.strAdapterStatus) - 1);
+  signalStatus.strAdapterStatus[sizeof(signalStatus.strAdapterStatus)-1] = '\0';
```
Could you please add blanks around the `-`?

fuzzard commented:
Cheers, done.
